src/qiboml/models/abstract.py

```diff
@@ -0,0 +1,67 @@
+"""Defines the general structure of a qiboml module"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+from qibo import Circuit
+from qibo.config import raise_error
+from qibo.gates import abstract
+
+from qiboml.backends import TensorflowBackend as JaxBackend
+
+
+@dataclass
+class QuantumCircuitLayer(ABC):
+
+    nqubits: int
+    qubits: list[int] = None
+    circuit: Circuit = None
+    initial_state: "ndarray" = None
+    backend: "qibo.backends.Backend" = JaxBackend()
+
+    def __post_init__(self) -> None:
+        if self.qubits is None:
+            self.qubits = range(self.nqubits)
+        self.circuit = Circuit(self.nqubits)
+
+    @abstractmethod
+    def forward(self, x):
+        pass
+
+    @abstractmethod
+    def backward(self, input_grad: "ndarray"):
+        pass
+
+    @property
+    def parameters(self) -> "ndarray":
+        return self.circuit.get_parameters()
+
+    @parameters.setter
+    def parameters(self, x: "ndarray") -> None:
+        self.circuit.set_parameters(x)
+
+
+@dataclass
+class SequentialQuantumModel(QuantumCircuitLayer):
+
+    layers: list[QuantumCircuitLayer] = None
```

Comment:
Are you sure we truly need a model for this? Can't we just stack circuits?

(for the time being, this is just a sequence of circuits - that is already a circuit on its own, so I'm not sure there is much benefit in having a dedicated object...)

Reply:
No, this was something that I did in the beginning, I am keeping it just for now, but once the ML interfaces are ready, this will be completely replaced by their own version of `Sequential`.